lib/rucio/client/downloadclient.py

```diff
@@ -1452,3 +1444,22 @@ def _send_trace(self, trace):
         """
         if self.tracing:
             send_trace(trace, self.client.host, self.client.user_agent)
+
+
+def _verify_checksum(item, path):
+    rucio_checksum = item.get(PREFERRED_CHECKSUM)
+    local_checksum = None
+    checksum_algo = CHECKSUM_ALGO_DICT.get(PREFERRED_CHECKSUM)
+
+    if rucio_checksum and checksum_algo:
+        local_checksum = checksum_algo(path)
+        return rucio_checksum == local_checksum, rucio_checksum, local_checksum
+
+    for checksum_name in GLOBALLY_SUPPORTED_CHECKSUMS:
+        rucio_checksum = item.get(checksum_name)
+        checksum_algo = CHECKSUM_ALGO_DICT.get(checksum_name)
+        if rucio_checksum and checksum_algo:
+            local_checksum = checksum_algo(path)
+            return rucio_checksum == local_checksum, rucio_checksum, local_checksum
+
+    return False
```

Comment:
The last return should return a tuple to have the function always returning the same type

Reply:
Done!